src/main/java/witchinggadgets/common/items/armor/ItemPrimordialArmor.java

```diff
@@ -147,19 +152,15 @@ public void onArmorTick(World world, EntityPlayer player, ItemStack stack) {
         }
 
         if (this.armorType == 3) {
-            if (!player.capabilities.isFlying && player.moveForward > 0.0F) {
-                if (player.worldObj.isRemote && !player.isSneaking()) {
-                    if (!Thaumcraft.instance.entityEventHandler.prevStep.containsKey(player.getEntityId()))
-                        Thaumcraft.instance.entityEventHandler.prevStep.put(player.getEntityId(), player.stepHeight);
-                    player.stepHeight = 1.0F;
-                }
-                if (player.onGround) {
-                    float bonus = 0.055F;
-                    if (player.isInWater()) bonus /= 4.0F;
-                    player.moveFlying(0.0F, 1.0F, bonus);
-                } else if (Hover.getHover(player.getEntityId())) player.jumpMovementFactor = 0.03F;
-                else player.jumpMovementFactor = 0.05F;
+            if (getIntertialState(stack) && player.moveForward == 0
+                    && player.moveStrafing == 0
+                    && player.capabilities.isFlying) {
+                player.motionX *= 0.5;
+                player.motionZ *= 0.5;
             }
+            float bonus = 0.7685F;
```

Comment:
This should be 0.055F for example

Reply:
i changed the speed bonus to be in par with the nano boots of the void speed, since this is a magical upgrade to the voidwalker boots, gated even later than the nano boots of the void themselves. So keeping this at normal walking speed would be way too slow.

Reply:
Absolutely fair then